### DISCUSSION TITLE
Search list defined with a model function

### DISCUSSION THREAD

**Discussion by rchampagneca** [Laravel-Backpack/community-forum#469] [Q&A (Help)]
Hello,

Having a hard time figuring this one. I've got a simple list using a model function : 
```php
//
// setupListOperation
//
        $this->crud->addColumn([
            'name' => 'employé.nom',
            'type'  => 'relationship',
            'label' => 'Nom'
        ]);
        $this->crud->addColumn([
            'name' => 'employé.prénom',
            'type' => 'relationship',
            'label' => 'Prénom'
        ]);

// Model function
//
    public function employé()
    {
        return $this->belongsTo(Employe::class);
    }

``` 
How can I get the **column search** to work ?

**Comment by tabacitu** [comment]
You can define your own search logic for columns using the `searchLogic` attribute. In your example it should be something like:

```php
        $this->crud->addColumn([
            'name' => 'employé.nom',
            'type'  => 'relationship',
            'label' => 'Nom',
            'searchLogic' => function ($query, $column, $searchTerm) {
                $query->whereHas('employe', function ($q) use ($column, $searchTerm) {
                    $q->where('nom', 'like', '%'.$searchTerm.'%');
                });
            }
        ]);
```

See https://backpackforlaravel.com/docs/5.x/crud-columns#custom-search-logic for more info.

Hope it helps!

**Reply by rchampagneca** [reply]
Hello @tabacitu ... yes, I had looked at [custom search logic](https://backpackforlaravel.com/docs/5.x/crud-columns#custom-search-logic-for-columns). But, the code below doesn't work :

```php
        $this->crud->addColumn([
            'name' => 'employé.nom',
            'type'  => 'text',
            'label' => 'Nom',
            'searchLogic' => function ($query, $column, $searchTerm) {
                $query->whereHas('employé', function ($q) use ($column, $searchTerm) {
                    $q->where('nom', 'like', '%' . $searchTerm . '%');
                });
            }
        ]);
```

**Reply by tabacitu** [reply]
Yes, you'll have to customize it for your particular use case. That example I have you might not work for you, it was just pseudocode 😉

You can dump the query after the WhereHas, or use DebugBar or Laravel Telescope to see if the query is what you want. You can also use something like Tinker (or Tinkerwell) to play around with your custom queries.
